modules/tinymce/src/plugins/table/test/ts/browser/ui/TableClassListButtonsTest.ts

```diff
@@ -0,0 +1,126 @@
+      TinyUiActions.clickOnToolbar(editor, 'button[title="Table styles"]');
+    } catch (e) {}
+  });
+
+  it('TINY-7476: Ensure that the checkmark appears for a single class', async () => {
+    const editor = hook.editor();
+
+    editor.setContent(
+      '<table>' +
+        '<tbody>' +
+          '<tr>' +
+            '<td>Filler</td>' +
+          '</tr>' +
+        '</tbody>' +
+      '</table>'
+    );
+
+    TinySelections.setSelection(editor, [ 0, 0, 0 ], 0, [ 0, 0, 0 ], 1);
+
+    const sugarContainer = SugarBody.body();
+
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Table styles"]');
+
+    await Waiter.pTryUntil('There should not be a checkmark', () => {
+      UiFinder.exists(sugarContainer, '.tox-menu');
+      UiFinder.exists(sugarContainer, '.tox-collection__item[aria-checked="false"]'); // No checked ones exist.
+      UiFinder.notExists(sugarContainer, '.tox-collection__item[aria-checked="true"]');
+    });
+
+    editor.execCommand('mceTableToggleClass', false, 'a');
+    assert.lengthOf(events, 1, 'Command executed successfully.');
+
+    // Cycle the menu to let the checkmark load.
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Table styles"]');
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Table styles"]');
+
+    await Waiter.pTryUntil('There should be a checkmark', () => {
+      UiFinder.exists(sugarContainer, '.tox-menu');
+      UiFinder.exists(sugarContainer, '.tox-collection__item[aria-checked="true"]'); // We have one checked (a) and one unchecked (b)
+      UiFinder.exists(sugarContainer, '.tox-collection__item[aria-checked="false"]');
+    });
+  });
+
+  it('TINY-7476: Ensure that the checkmark appears for two classes', async () => {
+    const editor = hook.editor();
+
+    editor.setContent(
+      '<table>' +
+        '<tbody>' +
+          '<tr>' +
+            '<td>Filler</td>' +
+          '</tr>' +
+        '</tbody>' +
+      '</table>'
+    );
+
+    TinySelections.setSelection(editor, [ 0, 0, 0 ], 0, [ 0, 0, 0 ], 1);
+
+    const sugarContainer = SugarBody.body();
+
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Table styles"]');
+
+    await Waiter.pTryUntil('There should not be a checkmark', () => {
+      UiFinder.exists(sugarContainer, '.tox-menu');
+      UiFinder.exists(sugarContainer, '.tox-collection__item[aria-checked="false"]'); // No checked ones exist.
+      UiFinder.notExists(sugarContainer, '.tox-collection__item[aria-checked="true"]');
+    });
+
+    editor.execCommand('mceTableToggleClass', false, 'a');
+    editor.execCommand('mceTableToggleClass', false, 'b');
+    assert.lengthOf(events, 2, 'Command executed successfully.');
+
+    // Cycle the menu to let the checkmark load.
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Table styles"]');
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Table styles"]');
+
+    await Waiter.pTryUntil('There should be a checkmark', () => {
```

Comment:
This doesn't really check what's mentioned in the description. It's only checking that 1 checkmark has appeared and that there's no others, it's possible that only 1 menu item rendered. It might be a good idea to use the `Assertions.assertPresence` API here instead which would allow you to assert there's 2 checkmarks.

- [ ] Make sure this asserts there's 2 checked menu items